### DISCUSSION TITLE
Is it possible to disallow manual triggers for a DAG ?

### DISCUSSION THREAD

**Discussion by pedro-cf** [apache/airflow#40973] [Q&A] [answered]
Is it possible to disallow manual triggers for a DAG ?

**Comment by omkar-foss** [comment, marked as answer]
Hey @pedro-cf, as of now there doesn't seem to be a direct way to do this in the Airflow UI. So I can think of the following way to handle this in the DAG:

In your DAG code (sample code below), you can check if it's an external trigger and if yes, throw an error to disallow the DAG from running further.

```python
from airflow.models.dagrun import DagRun
def sample_func(**context):
    dag_run: DagRun = context.get('dag_run')
    # Below condition will return true if DAG is triggered manually.
    if dag_run.external_trigger:
        raise RuntimeError('Manual DAG run disallowed.')

sample_operator = PythonOperator(
    task_id="sample_operator",
    python_callable=sample_func,
)
```

**Reply by pedro-cf** [reply]
Hey @omkar-foss  thanks, I assumed so and for now used this approach, and proceeded with a feature request: https://github.com/apache/airflow/issues/40990 :

```python
@task
def my_task(dag_run=None):
    if dag_run.external_trigger:
        raise AirflowFailException("Manual DAG run disallowed")
    ...
```
